src/backend/utils/cache/inval.c

```diff
@@ -792,10 +796,13 @@ AcceptInvalidationMessages(void)
  *		Initialize inval data for the current (sub)transaction.
  */
 static void
-PrepareInvalidationState(void)
+PrepareInvalidationState(bool isEnr)
 {
 	TransInvalidationInfo *myInfo;
 
+	if (transInvalInfo != NULL && !isEnr)
```

Comment:
how is this correct? transInvalInfo is transaction level structure while invalidation is relation specific. How can I mark local only in this structure where my transaction can have mix of both temp and regular tables?